app/services/prompt_context.py

```
@@ -0,0 +1,280 @@
+async def _fetch_url_source(source: Any) -> str:
+    import aiohttp
+
+    url = _get(source, "url")
+    _validate_fetch_url(url)
+    headers = _get(source, "headers") or {}
+    max_bytes = int(os.getenv("PROMPT_DATA_SOURCE_MAX_BYTES", DEFAULT_SOURCE_MAX_BYTES))
+
+    timeout = aiohttp.ClientTimeout(total=12)
+    try:
+        async with aiohttp.ClientSession(timeout=timeout) as session:
+            async with session.get(url, headers=headers, allow_redirects=False) as resp:
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)